FinancialTracker/ViewControllers/HomeViewController.swift

```diff
@@ -12,15 +12,30 @@ class HomeViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        // Do any additional setup after loading the view.
+        self.title = "Home"
+        self.navigationItem.setHidesBackButton(true, animated: true)
+        self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Sign Out", style: .plain, target: self, action: #selector(signOut))
     }
         
-    @IBAction func addButtonTapped(_ sender: Any) {
-        guard let expenseVC = self.storyboard?.instantiateViewController(withIdentifier: "ExpenseVC") as? ExpenseViewController else {
-            return
+    @objc func signOut() {
+        FirebaseHandler.shared.signOut { firebaseError, _ in
+            switch firebaseError {
+            case .signOut(let error):
+                guard let error = error else { return }
+                self.present(UIAlertController.create(title: "Sign Out Error", message: error.localizedDescription), animated: true)
+            case .database, .unknown, .access, .auth:
+                // swiftlint:disable:next force_unwrapping
+                assertionFailure("This error should not appear: \(firebaseError!.localizedDescription)")
+                // swiftlint:disable:next unneeded_break_in_switch
+                break
+            case .none:
+                break
```

Comment:
add some description here please

Reply:
As on the other cases it doesn't need one

Reply:
I am still not a fan of a `.none` hanging out by itself with no explanation at all. 

You can do this instead:

```swift
    @objc func signOut() {
        FirebaseHandler.shared.signOut { firebaseError, _ in
            switch firebaseError {
            case .signOut(let error):
                guard let error = error else { return }
                self.present(UIAlertController.create(title: "Sign Out Error", message: error.localizedDescription), animated: true)
            case .database, .unknown, .access, .auth:
                // swiftlint:disable:next force_unwrapping
                assertionFailure("This error should not appear: \(firebaseError!.localizedDescription)")
                // swiftlint:disable:next unneeded_break_in_switch
                break
            default:
                assertionFailure("\(firebaseError?.localizedDescription ?? "") not handled")
                break
            }
        }
    }
```

or this:

```swift
    @objc func signOut() {
        FirebaseHandler.shared.signOut { firebaseError, _ in
            switch firebaseError {
            case .signOut(let error):
                guard let error = error else { return }
                self.present(UIAlertController.create(title: "Sign Out Error", message: error.localizedDescription), animated: true)
            case .database, .unknown, .access, .auth, .none:
                // swiftlint:disable:next force_unwrapping
                assertionFailure("This error should not appear: \(firebaseError?.localizedDescription ?? "")")
                // swiftlint:disable:next unneeded_break_in_switch
                break
            }
        }
    }
```

or this:

```swift
  @objc func signOut() {
        FirebaseHandler.shared.signOut { firebaseError, _ in
            guard let firebaseError = firebaseError else {
                self.present(UIAlertController.create(title: "Sign Out Error", message: "Sign out failed with unknown error"), animated: true)
                return
            }
            switch firebaseError {
            case .signOut(let error):
                guard let error = error else { return }
                self.present(UIAlertController.create(title: "Sign Out Error", message: error.localizedDescription), animated: true)
            case .database, .unknown, .access, .auth:
                // swiftlint:disable:next force_unwrapping
                assertionFailure("This error should not appear: \(firebaseError.localizedDescription)")
                // swiftlint:disable:next unneeded_break_in_switch
                break
            }
        }
    }
```